jellyfinclient/src/components/LoginComponent.js

```diff
@@ -1,100 +1,96 @@
 import React, { Component } from 'react';
 import { Formik } from 'formik';
+import { connect } from 'react-redux';
 import {
     Button,
     Text,
     View,
     TextInput,
-    StatusBar    
+    StatusBar
 } from 'react-native';
 import styles from './Style'
-import { Link } from '../utilities/routing/index';
+import { Redirect } from '../utilities/routing'
+import JFInterface from '../actions/ApiClient';
 
-export default class LoginComponent extends Component {
+class LoginComponent extends Component {
     constructor(props, context) {
         super(props, context);
         this.state = {
-            message: 'Welcome to login component of this app.',
+            message: 'Login with your user credentials:',
             usernameMessage: 'Username: ',
             passwordMessage: 'Password: ',
-            serverAddressMessage: 'Server address: ',
-            serverAddress: '',
-            username: '',
-            password: '',
+            username: this.props.storage.authCredentials.username,
+            password: this.props.storage.authCredentials.password,
             loginButtonMessage: 'Login',
+            loginSuccess: false
         };
     }
 
+    componentDidUpdate() {
+        if (this.state.loginSuccess !== JFInterface.apiClient.isLoggedIn())
+            this.setState({ loginSuccess: JFInterface.apiClient.isLoggedIn() });
+    }
+
     render() {
-        return (            
-            <View style={styles.container}>
-                <StatusBar hidden/>
-                <Text style={[styles.text, styles.biggerText]}>
-                    {this.state.message}
-                </Text>
-                <Formik
-                    onSubmit={() => {
-                        this.props.loginAction(this.state)
-                    }}
-                    render={({
-                        handleSubmit,
-                    }) => (
-                            <View>
-                                <View style={styles.loginInput}>
-                                    <Text style={styles.text}>
-                                        {this.state.usernameMessage}
-                                    </Text>
+        return (
+            !JFInterface.apiClient ?
```

Comment:
This is not super readable. Would you mind splitting this into methods that return one template or the other?

Reply:
Updated